lib/app/index.js

```diff
@@ -1,53 +1,171 @@
-var async = require('async');
+'use strict';
 
-var mage = require('../mage');
-var logger = mage.core.logger;
+const memoize = require('memoizee');
 
 
 // app registration
 
-var appMap = {};
-var appList = [];
+const appMap = {};
+const appList = [];
 
 
 /**
- * This function creates instances of all configured apps and registers them into appMap and appList.
+ * Returns the configuration for all apps (overridden during unit tests)
+ *
+ * @returns {Object}
+ */
+/* istanbul ignore next */
+exports.getAppsConfig = memoize(function () {
+	return require('lib/mage').core.config.get(['apps']);
```

Comment:
How is this any different to just calling the function directly?
doesn't node cache the require, and doesn't the config system cache config entry?
Is it just to avoid the object lookups?